### PR TITLE
feat: add DeactivateDeal Tx keeper, and CLI

### LINKED ISSUE #530: Implement DeactivateDeal tx
- change Deal status
- graceful shutdown Deal (need discussion)
- refund remain budget to consumer

### PR DESCRIPTION
**Pull request by inchori**
Closes #530 

The branch name includes a `keeper`, but I just implemented cmd part, too. Because, this feature is not big as I thought.

### CODE REVIEW

File: x/datadeal/types/message_deal.go [hunk 1]
```diff
@@ -53,3 +53,45 @@ func (m *MsgCreateDeal) GetSigners() []sdk.AccAddress {
...
+	return "DeactivateDeal"
+}
+
+func (m *MsgDeactivateDeal) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(m.RequesterAddress)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid requester address (%s)", err)
+	}
+
+	if m.DealId == 0 {
```
Comment (audtlr24) [inline x/datadeal/types/message_deal.go:80]:
```suggestion
	if m.DealId <= 0 {
```

**Review by audtlr24 — APPROVED** [review]
Thanks, I left minor suggestion.

**Review by gyuguen — APPROVED** [review]
LGTM

File: x/datadeal/keeper/deal.go [hunk 2]
```diff
@@ -160,3 +160,43 @@ func (k Keeper) IncreaseCurNumDataOfDeal(ctx sdk.Context, dealID uint64) error {
...
 }
+
+func (k Keeper) DeactivateDeal(ctx sdk.Context, msg *types.MsgDeactivateDeal) error {
+	deal, err := k.GetDeal(ctx, msg.DealId)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrDeactivateDeal, err.Error())
+	}
+
+	if deal.ConsumerAddress != msg.RequesterAddress {
+		return sdkerrors.Wrapf(types.ErrDeactivateDeal, "requester address is invalid")
```
Comment (H4NLee) [inline x/datadeal/keeper/deal.go:171]:
```suggestion
		return sdkerrors.Wrapf(types.ErrDeactivateDeal, "only consumer can deactivate the deal")
```

**Review by H4NLee — APPROVED** [review]

**Review by youngjoon-lee — APPROVED** [review]
lgtm

**Pull request merged**